logic/strategy/src/one_hop_connection.rs

```diff
@@ -0,0 +1,358 @@
+use std::collections::HashMap;
+use std::fmt::{Debug, Display, Formatter};
+use std::ops::{Add, Sub};
+use std::str::FromStr;
+use tracing::{debug, error, info};
+use validator::Validate;
+
+use crate::errors::{self, Result};
+use crate::strategy::SingularStrategy;
+use crate::types::ChannelDecision;
+use crate::Strategy;
+
+#[cfg(all(feature = "prometheus", not(test)))]
+use hopr_metrics::metrics::SimpleCounter;
+
+#[cfg(all(feature = "prometheus", not(test)))]
+lazy_static::lazy_static! {
+    static ref METRIC_COUNT_OPENS: SimpleCounter =
+        SimpleCounter::new("hopr_strategy_one_hop_connection_opened_channels_count", "Count of open channel decisions").unwrap();
+    static ref METRIC_COUNT_CLOSURES: SimpleCounter =
+        SimpleCounter::new("hopr_strategy_one_hop_connection_closed_channels_count", "Count of close channel decisions").unwrap();
+}
+
+#[serde_as]
+#[derive(Debug, Clone, PartialEq, smart_default::SmartDefault, Validate, Serialize, Deserialize)]
+pub struct OneHopConnectionStrategyConfig {
+    /// Ethereum addresses of the destination peers.
+    #[validate(length(min = 1))]
+    #[serde_as(as = "Vec<DisplayFromStr>")]
+    pub destination_peers: Vec<Address>,
+
+    /// A quality threshold between 0 and 1 used to determine whether the strategy should open channel with the peer.
+    #[validate(range(min = 0_f64, max = 1.0_f64))]
+    #[default = 0.5]
+    pub network_quality_threshold: f64,
+
+    /// Specifies a minimum version (in semver syntax) of the peer the strategy should open a channel to.
+    ///
+    /// Default is ">=2.0.0"
+    #[serde_as(as = "DisplayFromStr")]
+    #[default(">=2.0.0".parse().expect("should be valid default version"))]
+    pub minimum_peer_version: semver::VersionReq,
+
+    /// Minimum stake that a channel's balance must not go below.
+    ///
+    /// Default is 1 HOPR
+    #[serde_as(as = "DisplayFromStr")]
+    #[default(Balance::new_from_str("1000000000000000000", BalanceType::HOPR))]
+    pub min_stake_threshold: Balance,
+
+    /// Funding amount.
+    ///
+    /// Defaults to 10 HOPR.
+    #[serde_as(as = "DisplayFromStr")]
+    #[default(Balance::new_from_str("10000000000000000000", BalanceType::HOPR))]
+    pub funding_amount: Balance,
+}
+
+/// This strategy is designed to maintain 1-hop pathes to certain destination peers,
+/// where there is at least one relay node that is not part of the destination peer set.
+///
+/// There should be outgoing channels to the destination peers.
+///
+/// When there's no relay node that is not part of the destination peer set,
+/// the strategy should select a relay node and open an outgoing channel to it.
+/// Selection of this new relay peer should prioritize those that have incoming channels
+/// with the destination peers.
+///
+/// It choses channels to close among the channels that cannot be used for reaching
+/// the destination peers with 1-hop.
+/// The strategy only outgoing channels where the target has low network quality.
```

Comment:
Closing the channels is a more problematic topic, than opening them, since it has a direct cost, but no direct benefit.